### PR TITLE
Java: RENAME 

### PR DESCRIPTION
**Pull request by cyip10**
https://redis.io/docs/latest/commands/rename/

### CODE REVIEW

File: java/client/src/main/java/glide/api/commands/GenericBaseCommands.java [hunk 1]
```diff
@@ -454,6 +454,25 @@ CompletableFuture<Boolean> pexpireAt(
      */
     CompletableFuture<Long> objectRefcount(String key);
 
+    /**
+     * Renames <code>key</code> to <code>newKey</code>.
+     *
+     * @apiNote If `<code>newKey</code> already exists it is overwritten. meaning that in practice
+     *     only keys that have the same hashtag can be reliably renamed in cluster.
```
Comment (aaron-congo) [inline java/client/src/main/java/glide/api/commands/GenericBaseCommands.java:461]: I would rephrase this a bit to be consistent with the Python RENAME docs and the other Java docs. The warning about the hash slot follows a consistent pattern so we should follow that pattern here as well
```suggestion
     * Renames <code>key</code> to <code>newKey</code>.<br>
     * If `<code>newKey</code> already exists it is overwritten.
     *
     * @apiNote  When in cluster mode, both <code>key</code> and <code>newKey</code> must map to the
     *     same hash slot.
```

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 2]
```diff
@@ -2704,6 +2705,21 @@ public T type(@NonNull String key) {
         return getThis();
     }
 
+    /**
+     * Renames <code>key</code> to <code>newKey</code>.
+     *
```
Comment (aaron-congo) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:2710]:
```suggestion
     * If `<code>newKey</code> already exists it is overwritten.
     *
```

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 3]
```diff
@@ -2704,6 +2705,21 @@ public T type(@NonNull String key) {
...
+     * @see <a href="https://redis.io/commands/rename/">redis.io</a> for details.
+     * @param key The <code>key</code> to rename.
+     * @param newKey The new name of the <code>key</code>.
+     * @return If the <code>key</code> was successfully renamed, return <code>"OK"</code>. If <code>
+     *     key</code> does not exist, an error is thrown.
```
Comment (aaron-congo) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:2715]: For the Transaction docs, we prefix the description with "Command Response - " since we are describing what the transaction returns when its executed rather than what the function itself returns
```suggestion
     * @return Command Response - If the <code>key</code> was successfully renamed, return <code>"OK"</code>. If <code>
     *     key</code> does not exist, the transaction fails with an error.
```

File: java/client/src/test/java/glide/api/RedisClientTest.java [hunk 4]
```diff
@@ -3894,6 +3895,27 @@ public void type_returns_success() {
...
 
+    @SneakyThrows
+    @Test
+    public void rename() {
+        // setup
+        String key = "key1";
+        String newKey = "key2";
+        String[] arguments = new String[] {key, newKey};
+        CompletableFuture<String> testResponse = new CompletableFuture<>();
+        testResponse.complete("OK");
```
Comment (aaron-congo) [inline java/client/src/test/java/glide/api/RedisClientTest.java:3906]: I think there is an imported constant for this already
```suggestion
        testResponse.complete(OK);
```

File: java/client/src/test/java/glide/api/RedisClientTest.java [hunk 5]
```diff
@@ -3894,6 +3895,27 @@ public void type_returns_success() {
...
+
+        // match on protobuf request
+        when(commandManager.<String>submitNewCommand(eq(Rename), eq(arguments), any()))
+                .thenReturn(testResponse);
+
+        // exercise
+        CompletableFuture<String> response = service.rename(key, newKey);
+
+        // verify
+        assertEquals(testResponse, response);
```
Comment (aaron-congo) [inline java/client/src/test/java/glide/api/RedisClientTest.java:3916]:
```suggestion
        assertEquals(testResponse, response);
        assertEquals(OK, response.get());
```

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 6]
```diff
@@ -1087,6 +1087,24 @@ public void smove(BaseClient client) {
...
+        assertEquals(1L, client.exists(new String[] {key1 + "_rename"}).get());
+
+        // key doesn't exist
+        ExecutionException executionException =
+                assertThrows(
+                        ExecutionException.class,
+                        () ->
+                                client.rename("{same_slot}" + "non_existing_key", "{same_slot}" + "_rename").get());
+    }
+
```
Comment (aaron-congo) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:1107]: Since this command is a multi-key command that does not allow passing keys with different hash tags, you will also have to add a line to cluster/CommandTests.java#callCrossSlotCommandsWhichShouldFail

File: java/client/src/main/java/glide/api/models/BaseTransaction.java (hunk 3 above)
Comment (cyip10) [reply java/client/src/main/java/glide/api/models/BaseTransaction.java:2715]: so right, i totally forgot about that

File: java/client/src/main/java/glide/api/commands/GenericBaseCommands.java [hunk 7]
```diff
@@ -454,6 +454,25 @@ CompletableFuture<Boolean> pexpireAt(
...
+     * @see <a href="https://redis.io/commands/rename/">redis.io</a> for details.
+     * @param key The key to rename.
+     * @param newKey The new name of the key.
+     * @return If the <code>key</code> was successfully renamed, return <code>"OK"</code>. If <code>
+     *     key</code> does not exist, an error is thrown.
+     * @example
+     *     <pre>{@code
+     * String value = client.set("key", "value").get();
+     * value = client.rename("key", "newKeyName");
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/GenericBaseCommands.java:470]:
```suggestion
     * value = client.rename("key", "newKeyName").get();
```

File: java/client/src/main/java/glide/api/commands/GenericBaseCommands.java (hunk 1 above)
Comment (Yury-Fridlyand) [reply java/client/src/main/java/glide/api/commands/GenericBaseCommands.java:461]: Agree with that. Please also remove dangling backtick `` ` ``.

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 8]
```diff
@@ -2704,6 +2705,22 @@ public T type(@NonNull String key) {
...
     }
 
+    /**
+     * If <code>newKey</code> already exists it is overwritten.
+     *
+     * @see <a href="https://redis.io/commands/rename/">redis.io</a> for details.
+     * @param key The <code>key</code> to rename.
+     * @param newKey The new name of the <code>key</code>.
+     * @return Command Response - If the <code>key</code> was successfully renamed, return <code>"OK"
+     *     </code>. If <code>
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:2715]: Please remove the line break

**Review by Yury-Fridlyand — APPROVED** [review]

File: java/integTest/src/test/java/glide/SharedCommandTests.java (hunk 6 above)
Comment (Yury-Fridlyand) [reply java/integTest/src/test/java/glide/SharedCommandTests.java:1107]: ^

File: java/client/src/main/java/glide/api/commands/GenericBaseCommands.java [hunk 9]
```diff
@@ -455,7 +455,27 @@ CompletableFuture<Boolean> pexpireAt(
...
+     *     <pre>{@code
+     * String value = client.set("key", "value").get();
+     * value = client.rename("key", "newKeyName").get();
+     * assert value.equals("OK");
+     * }</pre>
+     */
+    CompletableFuture<String> rename(String key, String newKey);
+
+    /**
+     * If <code>newKey</code> already exists it is overwritten.
```
Comment (aaron-congo) [inline java/client/src/main/java/glide/api/commands/GenericBaseCommands.java:478]: This line should remain unchanged since it belongs to a different command (renamenx)
```suggestion
     * Renames <code>key</code> to <code>newKey</code> if <code>newKey</code> does not yet exist.
```

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 10]
```diff
@@ -2704,6 +2705,22 @@ public T type(@NonNull String key) {
         return getThis();
     }
 
+    /**
+     * If <code>newKey</code> already exists it is overwritten.
```
Comment (aaron-congo) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:2709]:
```suggestion
     * Renames <code>key</code> to <code>newKey</code>.<br>
     * If <code>newKey</code> already exists it is overwritten.
```

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 11]
```diff
@@ -2704,6 +2705,22 @@ public T type(@NonNull String key) {
...
+     *
+     * @see <a href="https://redis.io/commands/rename/">redis.io</a> for details.
+     * @param key The <code>key</code> to rename.
+     * @param newKey The new name of the <code>key</code>.
+     * @return Command Response - If the <code>key</code> was successfully renamed, return <code>"OK"
+     *     </code>. If <code>
+     *     key</code> does not exist, the transaction fails with an error.
```
Comment (aaron-congo) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:2716]:
```suggestion
     *     </code>. If <code>key</code> does not exist, the transaction fails with an error.
```
